### ISSUE TITLE
Beat allow to pass build time version qualifier

### ISSUE THREAD

**Issue by ph** [elastic/beats#8186]
Allow passing a build time qualifier so that a build artifact can be promoted to GA without requiring a rebuild. Our build process accepts the `SNAPSHOT` environment variable to add the snapshot suffix, and we need to add a `VERSION_QUALIFIER` options that will allow the release manager to specify the qualifier to be used.

The generated package will use the specified qualifier and the beats will also report their version using the qualifier used at built time.

**Comment by ph** [comment]
Linked to APM https://github.com/elastic/apm-server/issues/1348

**Comment by graphaelli** [comment]
I'd like to request some feedback on https://github.com/elastic/apm-server/issues/1348#issuecomment-419563211, all of the changes made were in vendored beats code.

**Comment by tsg** [comment]
@graphaelli I saw your comment only after pushing this PR: https://github.com/elastic/beats/pull/8271. That is the result of Andrew walking me through the changes that are needed. 

Your PR seems to change the code version (e.g. changes in `GetDefaultVersion`), we've strictly altered the final package names, nothing in the contents of them.

**Comment by graphaelli** [comment]
Right, I've tried to handle both the code version and package name.  The tricky part was allowing each branch to set a default qualifier without requiring the build to provide one, or mage to know about the defaults - that is in 6.4.1 there is no qualifier by default, in 7.0.0 it's alpha1 unless otherwise specified.

**Comment by jasontedor** [comment]
We will not need default qualifiers as in the future all of our 7.0.0 snapshot builds will be unqualified. Qualifiers will only be used when preparing pre-releases.
